dotCMS/src/main/java/com/dotmarketing/portlets/folders/business/FolderAPIImpl.java

```diff
@@ -50,826 +51,838 @@
-		}
-		boolean move = folderFactory.move(folderToMove, newParentHost);
-
-		this.systemEventsAPI.pushAsync(SystemEventType.MOVE_FOLDER, new Payload(folderToMove, Visibility.EXCLUDE_OWNER,
-				new ExcludeOwnerVerifierBean(user.getUserId(), PermissionAPI.PERMISSION_READ, Visibility.PERMISSION)));
-
-		return move;
-	}
-
-	@CloseDBIfOpened
-	public List<Folder> findSubFolders(Host host, boolean showOnMenu) throws DotHibernateException{
-		return folderFactory.findSubFolders(host, showOnMenu);
-	}
-
-	@CloseDBIfOpened
-	public List<Folder> findSubFolders(Folder folder,boolean showOnMenu)throws DotStateException, DotDataException {
-		return folderFactory.findSubFolders(folder, showOnMenu);
-	}
-
-	@CloseDBIfOpened
-	public List<String> getEntriesTree(Folder mainFolder, String openNodes,String view, String content, String structureInode,User user)
-			throws DotStateException, DotDataException, DotSecurityException {
-		Locale locale = user.getLocale();
-		TimeZone timeZone = user.getTimeZone();
-		Role[] roles = (Role[])APILocator.getRoleAPI().loadRolesForUser(user.getUserId()).toArray(new Role[0]);
-		boolean isAdminUser = APILocator.getUserAPI().isCMSAdmin(user);
-		return folderFactory.getEntriesTree(mainFolder, openNodes, view, content, structureInode, locale, timeZone, roles, isAdminUser, user);
-	}
-
-
-	@CloseDBIfOpened
-	public List<String> getFolderTree(String openNodes, String view,
-			String content, String structureInode,User user)
-			throws DotStateException, DotDataException, DotSecurityException {
-		Locale locale = user.getLocale();
-		TimeZone timeZone = user.getTimeZone();
-		Role[] roles = (Role[])APILocator.getRoleAPI().loadRolesForUser(user.getUserId()).toArray(new Role[0]);
-		boolean isAdminUser = APILocator.getUserAPI().isCMSAdmin(user);
-		return folderFactory.getFolderTree(openNodes, view, content, structureInode, locale, timeZone, roles, isAdminUser, user);
-	}
-
-	@CloseDBIfOpened
-	public List<Contentlet> getLiveContent(Folder parent, User user,boolean respectFrontEndPermissions) throws DotDataException, DotSecurityException {
-		if (!permissionAPI.doesUserHavePermission(parent, PermissionAPI.PERMISSION_READ, user,respectFrontEndPermissions)) {
-			throw new DotSecurityException("User " + user.getUserId() != null?user.getUserId():"" + " does not have permission to read Folder " + parent.getName());
-		}
-		ChildrenCondition cond = new ChildrenCondition();
-		cond.live=true;
-		cond.deleted=false;
-		List list = folderFactory.getChildrenClass(parent, Contentlet.class, cond);
-
-		return permissionAPI.filterCollection(list, PermissionAPI.PERMISSION_READ, respectFrontEndPermissions, user);
-	}
-
-	@CloseDBIfOpened
-	public List<Link> getLiveLinks(final Folder parent, final User user,
-								   final boolean respectFrontEndPermissions)
-								   throws DotDataException, DotSecurityException {
-
-		if (!permissionAPI.doesUserHavePermission(parent, Permissi
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1166)